lib/cylc/gui/updater.py

```diff
@@ -135,174 +148,198 @@ def __init__(self, cfg, info_bar, restricted_display):
+            self.status = 'stopping'
+        elif glbl['paused']:
+            self.status = 'held'
+        elif glbl['will_pause_at']:
+            self.status = 'hold at ' + glbl[ 'will_pause_at' ]
+        elif glbl['will_stop_at']:
+            self.status = 'running to ' + glbl[ 'will_stop_at' ]
+        else:
+            self.status = 'running'
+        self.mode = glbl['run_mode']
+
+        if self.cfg.use_defn_order and 'namespace definition order' in glbl: 
+            # (protect for compat with old suite daemons)
+            nsdo = glbl['namespace definition order']
+            if self.ns_defn_order != nsdo:
+                self.ns_defn_order = nsdo
+                self.dict_ns_defn_order = dict(zip(nsdo, range(0,len(nsdo))))
+        try:
+            self.dt = get_time_string_from_unix_time(glbl['last_updated'])
+        except (TypeError, ValueError):
+            # Older suite...
+            self.dt = glbl['last_updated'].isoformat()
+        self.global_summary = glbl
 
-        if update_summaries:
-            try:
-                [glbl, states, fam_states] = self.god.get_state_summary()
-                self._retrieve_hierarchy_info() # may change on reload
-            except (Pyro.errors.ProtocolError, Pyro.errors.NamingError):
-                gobject.idle_add( self.connection_lost )
-                return False
+        if self.restricted_display:
+            states = self.filter_for_restricted_display(states)
 
-            if not glbl:
-                self.task_list = []
-                return False
+        self.full_state_summary = states
+        self.full_fam_state_summary = fam_states
+        self.refilter()
+
+    def get_stop_summary(self):
+        # Get the suite stop summary.
+        if self.stop_summary is not None and any(self.stop_summary):
+            self.info_bar.set_stop_summary(self.stop_summary)
+        else:
+            self.stop_summary = get_stop_state_summary(
+                self.cfg.suite, self.cfg.owner, self.cfg.host)
 
-            if glbl['stopping']:
-                self.status = 'stopping'
-            elif glbl['paused']:
-                self.status = 'held'
-            elif glbl['will_pause_at']:
-                self.status = 'hold at ' + glbl[ 'will_pause_at' ]
-            elif glbl['will_stop_at']:
-                self.status = 'running to ' + glbl[ 'will_stop_at' ]
+    def set_stopped(self):
+        self.connected = False
+        self.status = "stopped"
+        self.get_stop_summary()
+        self._summary_update_time = None
+        self.state_summary = {}
+        self.full_state_summary = {}
+        self.fam_state_summary = {}
+        self.full_fam_state_summary = {}
+        self.poll_schd.start()
+        self.info_bar.set_state([])
+        self.info_bar.set_status(self.status)
+        self._flag_new_update()
+
+    def update(self):
+        if cylc.flags.debug:
+            print >> sys.stderr, "UPDATE", ctime().split()[3],
```

Comment:
I think minimal timing info is fine here - this debug trace is just to allow developers to see what happens in what order when the gui connects, disconnects, and reconnects to a suite daemon.